attrs_strict/_type_validation.py

```diff
@@ -51,6 +72,25 @@ def _validate_elements(attribute, value, expected_type):
         _handle_tuple(attribute, value, expected_type)
     elif base_type == typing.Union:
         _handle_union(attribute, value, expected_type)
+    elif base_type in {typing.Callable, Callable}:
+        _handle_callable(attribute, value, expected_type)
+
+
+def _handle_callable(attribute, callable_, expected_type):
+    _signature = signature(callable_)
+    callable_args = [
+        param.annotation for param in _signature.parameters.values()
+    ]
+    callable_args.append(_signature.return_annotation)
+    expected_args = expected_type.__args__
+    if not expected_args:
+        return  # No annotations specified on type, matches all Callables
+
+    for callable_param, expected_param in zip_longest(
```

Comment:
what if `len(callable_param)` and `len(expected_param)` are diifferent ? Zip would only work for the shortest.

Reply:
`zip_longest` will put `None` for the case of uneven lengths. The check will fail and raise the error, which I think is correct behaviour since we only support fully annotated callables (or un-annotated ones) in the current implementation.